### DISCUSSION TITLE
HeadlessUI & Stimulus JS

### DISCUSSION THREAD

**Discussion by lb-** [tailwindlabs/tailwindcss#7546] [Ideas]
Has anyone got any recommendations for using Stimulus JS with Tailwind CSS at a general component level?

I'm thinking the headless UI package but ported / built for Stimulus. 

Any package recommendations or suggestions would be appreciated.

Alternatively, would there be an appetite for such a thing?

**Comment by fabstei** [comment]
There would definitely be appetite for a "stimulus port" of Headless UI. The lack of stimulus support is a major factor why we decided to stick with Vue.js for a small prototype we're currently building.

//Edit: No good recommendations so far, unfortunately

**Reply by lb-** [reply]
How would envisage the aria props / accessibility features to work?
 My thoughts are some kind of check / write to Dom on connect where any missing aria attributes are added. Then in debug mode this could throw an error to enforce initially correct aria attributes. 

I know there's more to accessibility than aria attributes but just thinking that this is something that Stimulus may look at differently vs react/Vue which controls the whole DOM.

**Reply by janbaykara** [reply]
That's an interesting thing to consider @lb-. In the same way that Adobe's [React-Aria](https://react-spectrum.adobe.com/react-aria/index.html) project provides 'primitives for accessible UI', could totally imagine Stimulus controllers which manage attrs and so on.

I'd definitely be in the market for HeadlessUI for Stimulus that comes with a TailwindCSS UI implementation.

**Comment by lb-** [comment]
Sorry but I didn't see that this had been asked previously on the HeadlessUi discussions. 

https://github.com/tailwindlabs/headlessui/discussions/824
https://github.com/tailwindlabs/headlessui/discussions/704

**Comment by mario-fehr** [comment]
A nice start https://github.com/excid3/tailwindcss-stimulus-components

**Comment by philliplongman** [comment]
Looks like that hasn't been updated in 8 months. Has there been any more thought given to this?
